### PR TITLE
fix: handle custom promo date format in firefox

### PR DESCRIPTION
**Pull request by firebomb1**
## Description

On Firefox promos are not displayed because we use a custom date format. `dayjs(String)` [only accepts ISO 8601](https://day.js.org/docs/en/parse/string). Chrome apparently is more lenient with String to date conversion so this bug wasn't noticed.

To support custom date format we can use the existing CustomParseFormat dayjs plugin.

This also include a fix to not return an empty Carousel when there is no rendered promo, which caused additional space above the TradeCard (in Chrome and Firefox)... which is the only reason I've noticed something was wrong in the first place (as dayjs doesn't seem to report invalid date parsing as errors) 😂

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

## Risk

None.

## Testing

Verify that the promo now appears on both Firefox and Chrome.
Possibly test without any current promo and notice

### Engineering
☝️ 

### Operations
☝️ 

## Screenshots (if applicable)
Before in Firefox:
image

After in Firefox: 
image

Chrome is unchanged.

The case when no promo is running (or all have expired), the UI issue mentioned in the description is now fixed in all browsers:
image

### CODE REVIEW

File: src/components/Promo/PromoCard.tsx [hunk 1]
```diff
@@ -11,6 +12,8 @@ import { WalletActions } from 'context/WalletProvider/actions'
 import { useWallet } from 'hooks/useWallet/useWallet'
 
 dayjs.extend(isBetween)
+dayjs.extend(customParseFormat)
+const promoDateFormat = 'YYYY-MM-DD hh:mm A'
```
Comment (gomesalexandre) [inline src/components/Promo/PromoCard.tsx:16]: 🧠

File: src/components/Promo/PromoCard.tsx [hunk 2]
```diff
@@ -95,10 +98,9 @@ export const PromoCard = () => {
       },
     )
   }, [handleClick])
-
-  return (
+  return renderPromos.length > 0 ? (
```
Comment (gomesalexandre) [inline src/components/Promo/PromoCard.tsx:101]: If you feel like clicking this button:

```suggestion
  return renderPromos.length ? (
```

**Review by gomesalexandre — APPROVED** [review]
Very nice catch @firebomb1 ! Tested in Chrome and found no regressions, confirmed this fixes the display issue in Firefox 🐐

**Review by 0xApotheosis — APPROVED** [review]
Nice fix @firebomb1 🔥 

Gomes' trivial suggestion was addressed 👌 good to go.

**Pull request merged**